### PR TITLE
Fix crash when trying to resolve a generic method that doesn't exist in the later compilation.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/17056

### CODE REVIEW

File: src/Workspaces/Core/Portable/SymbolKey/SymbolKey.MethodSymbolKey.cs [hunk 1]
```diff
@@ -144,8 +144,13 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader)
                 {
                     // We didn't find any candidates.  We still need to stream through this
                     // method signature so the reader is in a proper position.
+
+                    // Push an null-method to our stack so that any method-type-parameters
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/SymbolKey/SymbolKey.MethodSymbolKey.cs]: "an null"

File: src/Workspaces/CoreTest/SymbolKeyTests.cs [hunk 2]
```diff
@@ -579,6 +579,33 @@ void foo()
...
+";
+
+            var compilation1 = GetCompilation(source1, LanguageNames.CSharp);
+            var compilation2 = GetCompilation(source2, LanguageNames.CSharp);
+
+            var methods = GetDeclaredSymbols(compilation1).OfType<IMethodSymbol>();
+            foreach (var method in methods)
+            {
+                var key = SymbolKey.Create(method);
+                key.Resolve(compilation2);
```
Comment (Pilchie) [inline src/Workspaces/CoreTest/SymbolKeyTests.cs]: Should you assert something about the resolution?

Comment (CyrusNajmabadi) [inline src/Workspaces/CoreTest/SymbolKeyTests.cs]: Sure.  i can do that.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Pull request merged**